sonarr/series.go

```diff
@@ -79,61 +181,77 @@ func (s *Sonarr) AddSeriesContext(ctx context.Context, series *AddSeriesInput) (
+func (s *Sonarr) GetSeriesByID(seriesID int64) (*SeriesOutput, error) {
 	return s.GetSeriesByIDContext(context.Background(), seriesID)
 }
 
-func (s *Sonarr) GetSeriesByIDContext(ctx context.Context, seriesID int64) (*Series, error) {
-	var series Series
+func (s *Sonarr) GetSeriesByIDContext(ctx context.Context, seriesID int64) (*SeriesOutput, error) {
+	var output SeriesOutput
 
-	_, err := s.GetInto(ctx, "v3/series/"+strconv.FormatInt(seriesID, starr.Base10), nil, &series)
-	if err != nil {
-		return nil, fmt.Errorf("api.Get(series): %w", err)
+	uri := path.Join(bpSeries, strconv.Itoa(int(seriesID)))
+	if _, err := s.GetInto(ctx, uri, nil, &output); err != nil {
+		return nil, fmt.Errorf("api.Get(%s): %w", uri, err)
 	}
 
-	return &series, nil
+	return &output, nil
 }
 
 // GetSeriesLookup searches for a series [in Servarr] using a search term or a tvdbid.
 // Provide a search term or a tvdbid. If you provide both, tvdbID is used.
-func (s *Sonarr) GetSeriesLookup(term string, tvdbID int64) ([]*Series, error) {
+func (s *Sonarr) GetSeriesLookup(term string, tvdbID int64) ([]*SeriesOutput, error) {
 	return s.GetSeriesLookupContext(context.Background(), term, tvdbID)
 }
 
-func (s *Sonarr) GetSeriesLookupContext(ctx context.Context, term string, tvdbID int64) ([]*Series, error) {
-	params := make(url.Values)
+func (s *Sonarr) GetSeriesLookupContext(ctx context.Context, term string, tvdbID int64) ([]*SeriesOutput, error) {
+	var output []*SeriesOutput
 
+	params := make(url.Values)
 	if tvdbID > 0 {
 		params.Add("term", "tvdbid:"+strconv.FormatInt(tvdbID, starr.Base10))
 	} else {
 		params.Add("term", term)
 	}
 
-	var series []*Series
-
-	_, err := s.GetInto(ctx, "v3/series/lookup", params, &series)
-	if err != nil {
-		return nil, fmt.Errorf("api.Get(series/lookup): %w", err)
+	uri := path.Join(bpSeries, "lookup")
+	if _, err := s.GetInto(ctx, uri, params, &output); err != nil {
+		return nil, fmt.Errorf("api.Get(%s): %w", uri, err)
 	}
 
-	return series, nil
+	return output, nil
 }
 
 // Lookup will search for series matching the specified search term.
 // Searches for new shows on TheTVDB.com utilizing sonarr.tv's caching and augmentation proxy.
-func (s *Sonarr) Lookup(term string) ([]*Series, error) {
+func (s *Sonarr) Lookup(term string) ([]*SeriesOutput, error) {
 	return s.LookupContext(context.Background(), term)
 }
 
-func (s *Sonarr) LookupContext(ctx context.Context, term string) ([]*Series, error) {
+func (s *Sonarr) LookupContext(ctx context.Context, term string) ([]*SeriesOutput, error) {
 	return s.GetSeriesLookupContext(ctx, term, 0)
 }
+
+// DeleteSeries removes a single Series.
+func (s *Sonarr) DeleteSeries(seriesID int) error {
+	return s.DeleteSeriesContext(context.Background(), seriesID)
+}
+
+func (s *Sonarr) DeleteSeriesContext(ctx context.Context, seriesID int) error {
+	params := make(url.Values)
+	params.Add("deleteFiles", "true")
+	params.Add("addImportListExclusion", "false")
```

Comment:
@austinwbest quote from OP: "I enforced as general behaviour to remove files and not exclude from importlist. If you prefer, we can pass the two flags as inputs."

What do you think about this being default behavior? Just looking for more opinions.

Reply:
We can surely set them as input! If you want we can also have another method `DeleteSeriesDefault` or something like that with default values.

Reply:
I didn't hear from Austin (forgot to ping him in Discord today), but I do like the idea of making these inputs.
Something like:
```
func (s *Sonarr) DeleteSeriesContext(ctx context.Context, seriesID int, deleteFiles, importExclude bool) error {
```

I just know someone is going to ask for this, so we may as well provide it. Add a "comment" to explain what the inputs do please. Thanks!